controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -323,6 +323,65 @@ function readInputData() {
 		));
 	}
 
+	/**
+	 * @copydoc Form::validate()
+	 */
+	function validate($callHooks = true) {
+		// Display error if anonymity is impacted, i.e. if:
+		// 1) several blind reviewers are selected, or
+		// 2) a blind reviewer and another participant (other than editor) are selected.
```

Comment:
I think it should be possible for an assistant assigned to the correct stage to see a blind reviewer...

Reply:
Is there a possibility to ensure that they are assigned to the correct stage (if it is configurable)? Or can I assume that this is a case and treat them as editors?

Reply:
Sorry for bothering you for one more thing: Actually, a participant can either be a reviewer or have another role for a submission and all stages -- i. e. a reviewer cannot be assigned e. g. as proofreader as well, or? -- I do not know if the system checks it, but it wouldn't make sense, or?
If that could be possible, currently (as I implemented it) the proofreader (that was also a reviewer) could not participate in a discussion together with e. g. author in the production stage. Correct so?

Reply:
I guess if a user any stage assignment, they should be able to speak to authors. But only stage assignments in the same review stage should be able to speak to reviewers. (So in OMP we'll need to distinguish between internal/external review stages.)

Maybe the thing to do is to get the discussion's stage, then get a list of all stage assignments, reviewers and authors specific to that stage. And then to apply the logic which checks discussion assignments against these specific lists?